src/features/validation/index.ts

```diff
@@ -0,0 +1,241 @@
+import type Ajv from 'ajv';
+import type { JSONSchema7 } from 'json-schema';
+
+import type { IApplicationMetadata } from 'src/features/applicationMetadata';
+import type { IAttachments } from 'src/features/attachments';
+import type { Expression, ExprValToActual } from 'src/features/expressions/types';
+import type { TextReference, ValidLangParam } from 'src/features/language/useLanguage';
+import type { Visibility } from 'src/features/validation/visibility';
+import type { ILayoutSets } from 'src/layout/common.generated';
+import type { IInstance, IProcess } from 'src/types/shared';
+import type { LayoutNode } from 'src/utils/layout/LayoutNode';
+
+export enum FrontendValidationSource {
+  EmptyField = '__empty_field__',
+  Schema = '__schema__',
+  Component = '__component__',
+  Expression = '__expression__',
+}
+
+export type ValidationSeverity = 'error' | 'warning' | 'info' | 'success';
+
+export enum ValidationIssueSources {
+  File = 'File',
+  ModelState = 'ModelState',
+  Required = 'Required',
+  Expression = 'Expression',
+  Custom = 'Custom',
+}
+
+export enum BackendValidationSeverity {
+  Error = 1,
+  Warning = 2,
+  Informational = 3,
+  Success = 5,
+}
+
+// prettier-ignore
+export enum ValidationMask {
+  Schema                = 0b0000000000000001,
+  Component             = 0b0000000000000010,
+  Expression            = 0b0000000000000100,
+  CustomBackend         = 0b0000000000001000,
+  Required              = 0b0100000000000000,
+  AllExceptRequired     = 0b0011111111111111, // All frontend validations except required
+  All                   = 0b0111111111111111, // All frontend validations
+  Backend               = 0b1000000000000000, // All backend validations except custom backend validations
+  All_Including_Backend = 0b1111111111111111, // All validations including backend validations that overlap with frontend validations
+}
```

Comment:
Why the different naming schemes? I would expect every key to use PascalCase, but the last one here uses a combination of pascal and snake casing.

Reply:
Wow, I totally agree 🤦